app/grandchallenge/participants/views.py

```diff
@@ -107,3 +111,74 @@ def get_success_url(self):
             "participants:registration-list",
             kwargs={"challenge_short_name": self.object.challenge.short_name},
         )
+
+
+class RegistrationQuestionList(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, ListView
+):
+    model = RegistrationQuestion
+    permission_required = "change_challenge"
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+
+    def get_permission_object(self):
+        return self.request.challenge
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        queryset = queryset.filter(Q(challenge=self.request.challenge))
+        return queryset
+
+
+class RegistrationQuestionMixin(
+    LoginRequiredMixin,
+    SuccessMessageMixin,
+    ObjectPermissionRequiredMixin,
+):
+    model = RegistrationQuestion
+
+    permission_required = "change_challenge"
```

Comment:
You have a security bug here, another challenge admin could Update or Delete questions from other challenges if they knew the UUIDs as you're only inspecting the request and not the objects that you're editing. 

It is safer to:

- On the challenge: create an `add_registration_question` object permission on the challenge, assign that to the editors group.
- For the update and delete views use the existing `change_registrationquestion` and `delete_registrationquestion` object based permissions. Creating new tables from `UserObjectPermissionBase` and `GroupObjectPermissionBase` would be required.

Honestly I do not think a mixin is worth it here.

Reply:
I sort of like the usage of:
```Python
    def get_permission_object(self):
        return self.request.challenge
```

When checking `"challenges.add_registration_question"` permission against the challenge itself in the creation view. Since you propose to remove (or ban) the use of the function. I agree that it smells. What would be an acceptable alternative? Implement the check ourselves in the `dispatch()`, directly? As follows:

```Python
def dispatch(self, request, *args, **kwargs):
        if not self.request.user.has_perm(
            "challenges.add_registration_question",
            self.request.challenge,
        ):
            raise PermissionDenied
        return super().dispatch(request, *args, **kwargs)
```

Reply:
Maybe the better approach would be to ban the usage of the function `get_permission_object` with an exception for `CreateView` derivatives.

Reply:
Yes it is necessary for the create view as you need an object to test against, in this case returning the challenge is correct but the permission should be "add_registration_question".